numpy/core/src/multiarray/ctors.c

```diff
@@ -1025,7 +1027,7 @@ PyArray_NewFromDescr_int(PyTypeObject *subtype, PyArray_Descr *descr, int nd,
              * the memory, but be careful with this...
              */
             memcpy(fa->strides, strides, sizeof(npy_intp)*nd);
-            sd *= size;
+            sd = size;
```

Comment:
One small concern: we are redoing the calculation of `size` in `_array_fill_strides` a few lines above this. Maybe it is worth making that function `void` and moving this assignment to after the `if/else` for both branches? I would also add a comment, either  to `_array_fill_strides` or right before calling it, saying that the product of the `dims` and `sd` should already have been checked for overflow when calling it, but that may just be me being paranoid...